src/cmap/test/event.rs

```diff
@@ -1,29 +1,27 @@
-use std::sync::{Arc, RwLock};
+use std::{
+    collections::VecDeque,
+    sync::{Arc, RwLock},
+};
 
 use serde::{de::Unexpected, Deserialize, Deserializer};
 
 use crate::{event::cmap::*, options::StreamAddress};
 
 #[derive(Clone, Debug)]
 pub struct EventHandler {
-    ignore: Vec<String>,
-    pub events: Arc<RwLock<Vec<Event>>>,
+    pub events: Arc<RwLock<VecDeque<Event>>>,
```

Comment:
Is there a reason this was switched from Vec to VecDeque? I can't find any changes below that seem like they'd require it.

Reply:
It's to allow for quicker popping from the front when iterating over the events in order. In fact, we do the same when we purge connections, so I've updated the `available_connections` to be a `VecDeque` as well.

Reply:
This isn't necessary for the events anymore now that you switched to using `drain` though, right?

Reply:
We don't actually even need the `drain` there, we can just iterate over the events without removing any. Reverted to regular `Vec`.